pkg/cloudprovider/aws/suite_test.go

```diff
@@ -98,18 +118,19 @@ var _ = Describe("Allocation", func() {
 	var provisioner *v1alpha3.Provisioner
 
 	BeforeEach(func() {
-		provisioner = &v1alpha3.Provisioner{
+		var err error
+		provisioner, err = (&v1alpha3.Provisioner{
 			ObjectMeta: metav1.ObjectMeta{
 				Name: v1alpha3.DefaultProvisioner.Name,
 			},
 			Spec: v1alpha3.ProvisionerSpec{
 				Cluster: v1alpha3.Cluster{
 					Name:     ptr.String("test-cluster"),
 					Endpoint: "https://test-cluster",
-					CABundle: ptr.String("dGVzdC1jbHVzdGVyCg=="),
 				},
 			},
-		}
+		}).WithDynamicDefaults(ctx)
```

Comment:
thoughts on just isolating this to the validation?

Reply:
Why would that be better? In practice all this code under test is going to happen after defaulting has occurred. But I could be missing something.

Reply:
You're right I was confused. 

In general -- I don't like this dynamic defaulting behavior, since it results in an extra file read per provisioner per reconciliation and adds a nontrivial amount of complexity. Perhaps not a huge deal. 

I wonder if we could get away with statically defaulting it into the provisioner spec (in the Default() webhook), since AFAICT clusters don't typically rotate their root CA bundle. Even if one did, you'd simply need to apply the provisioner again and it would regenerate the new bundle.

Reply:
I suspect the OS level performance impact would be minimal, presumably that small of a file would stick around in memory if it were being accessed frequently and I think the overhead of a system call won't be too much given the other stuff we're already doing. 

I'm a little leery of the putting it into the static defaulter. While I agree that it seems mostly better/cleaner, I'm worrying about the case you're outlining where one would have to reapply the Provisioner (that seems slightly yucky), even if we don't know of a case where that happens today (I assume people will want to rotate those creds at some point, and it might be kind of a bummer to diagnose Karpenter issues as a result of a seemingly unrelated action). Also it just seems like the spirit of:

https://kubernetes.io/docs/tasks/run-application/access-api-from-pod/#directly-accessing-the-rest-api

is to honor these files under `/var` even if it is a bit fiddly?

Reply:
Ack